### PR TITLE
Pin all the things

### PR DESCRIPTION
**Pull request by Dreamsorcerer**
Pin all the things, to help stop CI breaking every week.

### CODE REVIEW

File: demos/imagetagger/pytest.ini [hunk 1]
```diff
@@ -11,7 +11,10 @@ addopts =
 
     # show values of the local vars in errors:
     --showlocals
+asyncio_mode = auto
 filterwarnings =
     error
+    # Not sure what is causing this warning...
```
Comment (webknjaz) [inline demos/imagetagger/pytest.ini:17]: Can't find it right now but I saw in the diff somewhere above, a BytesIO object being constructed inline as an argument to some PIL invocation. That's your unclosed file descriptor. It should be fixed by using a context manager there.

Comment (Dreamsorcerer) [reply demos/imagetagger/pytest.ini]: Unfortunately, that's not the one. Still get the same warning.

Comment (webknjaz) [reply demos/imagetagger/pytest.ini:17]: @Dreamsorcerer that is the one, you just applied the wrong fix. You're making sure to close the image object but not the underlying `BytesIO` object that is what causing the warnings in the first place.

Comment (webknjaz) [reply demos/imagetagger/pytest.ini:17]: Do this: https://github.com/aio-libs/aiohttp-demos/pull/196/files/97969478c8ecded46ca45b3bf50c53e3b863ee7a#r1009870281.

File: demos/imagetagger/imagetagger/worker.py [hunk 2]
```diff
@@ -46,9 +46,9 @@ def predict(raw_data: bytes, model: Optional[Any] = None) -> bytes:
     if model is None:
         raise RuntimeError('Model should be loaded first')
 
-    image = Image.open(io.BytesIO(raw_data))
     data: Dict[str, Any] = {}
-    image = prepare_image(image, target=(224, 224))
+    with Image.open(io.BytesIO(raw_data)) as f:
+        image = prepare_image(f, target=(224, 224))
```
[the remark this answers is not in the document]
Comment (Dreamsorcerer) [reply demos/imagetagger/imagetagger/worker.py]: Somehow you seem to be commenting on an outdated diff...
I already made that change: https://github.com/aio-libs/aiohttp-demos/pull/196/files#diff-bea02dda5f6c37a9c3db685f1227b5f7034f52e6545e66079cc05a58a7480582R50
Unfortunately, the warning still occurred.

The BufferedRandom part suggests to me that it is not `BytesIO`, but presumably this class:
https://docs.python.org/3.8/library/io.html#io.BufferedRandom
So, I think this must be getting created in some other part of the code.

[the remark this answers is not in the document]
Comment (webknjaz) [reply demos/imagetagger/imagetagger/worker.py]: Can it be that you forgot to commit the change or reverted it somehow? GH doesn't think this is outdated.

[the remark this answers is not in the document]
Comment (webknjaz) [reply demos/imagetagger/imagetagger/worker.py]: Alright, GH still shows this as a non-outdated thread, but I see, having checked from the laptop, that it's indeed committed.

Sometimes, GitHub's eventual consistency creates weird situations. Occasionally, it happens that a PR branch head pointer doesn't get updated and integrations, like CIs, keep testing against an outdated commit. It usually helps to close/reopen a PR and/or force-push the branch, so it gets reset. If it doesn't help, maybe it's a good idea to create a new branch and a PR.

P.S. This PR has so many changes that it's easy to get lost. So I'd recommend spitting out the formatting and functional changes to make it of a digestible size.

**Comment by webknjaz** [comment]
> The BufferedRandom part suggests to me that it is not `BytesIO`, but presumably this class:
> [docs.python.org/3.8/library/io.html#io.BufferedRandom](https://docs.python.org/3.8/library/io.html#io.BufferedRandom)
> So, I think this must be getting created in some other part of the code.

@Dreamsorcerer now that you mentioned `BufferedRandom`, I remember where I saw it in the past. It's created in the requests that receive files. I fixed it on aiohttp master via https://github.com/aio-libs/aiohttp/pull/5494/files#diff-c26d26819e57baa8710a1679bd0a660981b94b8ad1361822ec25009473994f09R825-R833 but on older branches, it's not possible to do the same.
So, in tests, I just resorted to closing these file handles in the request handlers:
- https://github.com/aio-libs/aiohttp/blob/master/tests/test_client_functional.py#L1247-L1248
- https://github.com/aio-libs/aiohttp/blob/master/tests/test_web_functional.py#L336

You could do the same here. The request originating in https://github.com/aio-libs/aiohttp-demos/blob/0f47a8a/demos/imagetagger/tests/test_views.py#L18 seems to leave a hanging file descriptor on the back-end. It is accessed here https://github.com/aio-libs/aiohttp-demos/blob/0f47a8aa3524758b7cc36c3592954f322567dce6/demos/imagetagger/imagetagger/views.py#L24 and so this is where you can close it after reading, once it's no longer needed.

**Comment by Dreamsorcerer** [comment]
@webknjaz Does the ignore in pytest.ini not work in conftest.py? Not sure why I'm still getting a failure with this DeprecationWarning, unless I'm missing something obvious.

**Comment by webknjaz** [comment]
@Dreamsorcerer it'd be easier if you could show this happening in CI. It's hard to say what you're observing.

**Comment by Dreamsorcerer** [comment]
That should be in the CI. The test failure I was seeing in the last test run was the
warning I added to the ignore for moderator_bot.

**Comment by Dreamsorcerer** [comment]
```
ImportError while loading conftest '/home/runner/work/aiohttp-demos/aiohttp-demos/demos/moderator_bot/tests/conftest.py'.
demos/moderator_bot/tests/conftest.py:2: in <module>
    from moderator_bot.server import init_application
demos/moderator_bot/moderator_bot/server.py:7: in <module>
    from aioslacker import Slacker
/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/aioslacker/__init__.py:18: in <module>
    class BaseAPI(slacker.BaseAPI):
/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/aioslacker/__init__.py:49: in BaseAPI
    def __request(self, method, api, **kwargs):
/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/asyncio/coroutines.py:111: in coroutine
    warnings.warn('"@coroutine" decorator is deprecated since Python 3.8, use "async def" instead',
E   DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
```

**Comment by Dreamsorcerer** [comment]
Nevermind, just didn't understand the matches are from the start of the string, so won't match the middle. Probably uses re.match(), rather than re.search() or similar.

**Comment by webknjaz** [comment]
@Dreamsorcerer it's described here https://docs.python.org/3/library/warnings.html#the-warnings-filter.
One thing that isn't is that you have to replace some characters of the message in the filter string, like `:` — the separator that you can't escape, with a match-all dot. Same with parens, I think.

**Pull request merged**